File: PY/Cristian Maldonado/ExponentialSearch.py
def BinarySearch(arr, inicio, fin, numerobuscado):
    if(fin>=inicio):
        medio = int((inicio+(fin-1))/2)
        if(arr[medio] == numerobuscado):
            return medio
        if(arr[medio] > numerobuscado):
            return BinarySearch(arr,inicio,medio-1,numerobuscado)
        else:
            return BinarySearch(arr,medio+1,fin,numerobuscado)
    return -1

def exponentialSearch(arr,tam,buscado):
    if(arr[0] == buscado):
        return 0
    i = int(1)
    while (i<tam and arr[i]<=buscado):
        i = i*2
    return BinarySearch(arr,i//2,min(i,tam),buscado)

lista = [1,2,3,4,5,6]
buscado = int(6)
resultado = int(exponentialSearch(lista, len(lista), buscado))
if(resultado < 0):
    print ("El elemento no se encuentra dentro del arreglo")
else:
    print("El elemento esta en la posicion:",resultado)

File: Java/Cristian Maldonado/Exponential_Search.java
import java.util.ArrayList;
import java.util.Arrays;

public class Exponential_Search {

	public static void main(String[] args) {
		ArrayList<Integer> lista = new ArrayList<Integer>();
		lista.add(1);
		lista.add(2);
		lista.add(3);
		lista.add(4);
		lista.add(5);
		int numerobuscado = 1;
		Integer arr[] = new Integer [lista.size()];
		int resultado = exponentialSearch(lista.toArray(arr) ,lista.size(), numerobuscado);
		System.out.println(resultado < 0 ? "El elemento no esta presente en el array" : 
			"El elemento esta en la posicion: "+resultado);
	}
	
	static int exponentialSearch(Integer [] arr, int n, int x) {
		if(arr [0] == x) {
			return 0;
		}
		int i = 1;
		while(i<n&&arr[i]<=x) {
			i = i*2;
		}
			return Arrays.binarySearch(arr,i/2,Math.min(i, n), x);
	}
}
